docs/src/content/en/docs/workflows/control-flow.mdx

```diff
@@ -1,265 +1,310 @@
 
-export const testWorkflow = createWorkflow({...})
-.dountil(nestedWorkflowStep, async ({ inputData: { userResponse } }) => userResponse === "yes")
-.commit();
-```
+### Looping with `.dountil()`
 
-#### Limiting loops
+Use `.dountil()` to run a step repeatedly until a condition becomes true.
 
-The `iterationCount` tracks how many times the loop step has run. You can use this to limit the number of iterations and prevent infinite loops. Combine it with `inputData` values to stop the loop after a set number of attempts.
+![Repeating with .dountil()](/image/workflows/workflows-control-flow-dountil.jpg)
 
-```typescript {7} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
-import { createWorkflow, createStep } from "@mastra/core/workflows";
-import { z } from "zod";
+```typescript {17} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
+const step1 = createStep({...});
 
-const counterStep = createStep({...});
+const step2 = createStep({
+  // ...
+  execute: async ({ inputData }) => {
+    const { number } = inputData;
+    return {
+      number: number + 1
+    };
+  }
+});
 
-export const testWorkflow = createWorkflow({...})
-.dountil(nestedWorkflowStep, async ({ inputData: { userResponse, iterationCount } }) => userResponse === "yes" || iterationCount >= 10)
-.commit();
+export const testWorkflow = createWorkflow({
+  // ...
+})
+  .then(step1)
+  .dountil(step2, async ({ inputData: { number } }) => number > 10)
+  .commit();
 ```
 
-#### Aborting loops
+### Looping with `.dowhile()`
 
-Use `iterationCount` to limit how many times a loop runs. If the count exceeds your threshold, throw an error to fail the step and stop the workflow.
+Use `.dowhile()` to run a step repeatedly while a condition remains true.
 
-```typescript {7} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
-import { createWorkflow, createStep } from "@mastra/core/workflows";
-import { z } from "zod";
+![Repeating with .dowhile()](/image/workflows/workflows-control-flow-dowhile.jpg)
 
-const counterStep = createStep({...});
+```typescript {17} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
+const step1 = createStep({...});
 
-export const testWorkflow = createWorkflow({...})
-.dountil(nestedWorkflowStep, async ({ inputData: { userResponse, iterationCount } }) => {
-  if (iterationCount >= 10) {
-    throw new Error("Maximum iterations reached");
+const step2 = createStep({
+  // ...
+  execute: async ({ inputData }) => {
+    const { number } = inputData;
+    return {
+      number: number + 1
+    };
   }
-  return userResponse === "yes";
+});
+
+export const testWorkflow = createWorkflow({
+  // ...
 })
-.commit();
+  .then(step1)
+  .dowhile(step2, async ({ inputData: { number } }) => number < 10)
+  .commit();
 ```
 
-### Repeating with `.foreach()`
+### Looping with `.foreach()`
```

Comment:
mention forEach concurrency, and how it differs from parallel? what is the default concurrency btw